Clean up summarize doc (#4089)
brimdata/zed: docs/language/operators/summarize.md

### Synopsis

```

> summarize [<field>:=]<agg> [where <expr>][, [<field>:=]<agg> [where <expr>] ...] [by [<field>][:=<expr>] ...]

Author: I figure the `...` should be added because the user is free to keep tacking on as many aggregate function calls as they want.

Also, I briefly thought about changing `summarize` to `[summarize]` to emphasize its optional nature due to the "implied operator" concept. However, I wasn't sure if we could say it's _always_ optional or if there's times it's necessary to remove ambiguity that I can't think of, so I've left that one as is. But please speak up if you do feel it should be `[summarize]`.

Author: @nwt: Ah, thanks for the pointer to precedent! Pushed a commit going with `[summarize]`.

```
### Description

The `summarize` operator consumes all of its input, applies an [aggregate function](../aggregates/README.md)
to each input value optionally organized with the group-by keys specified after
the `by` keyword, and at the end of input produces one or more aggregations
for each unique set of group-by key values.

> The keyword `summarize` can be treated as an [implied operator](../overview.md#26-implied-operators)
> and hence may be omitted before an aggregate function call.

Reviewer: Nit: Copy the corresponding language that appears in put.md.
```suggestion
The `summarize` keyword is optional since it is an
[implied operator](../overview.md#26-implied-operators).
```

> Each aggregate function may be optionally followed by a [`where`](where.md) clause, which

Author: Are folks ok with me linking to the `where` operator doc from here? I don't know if it's technically 100% the same thing since it's being used to affect the behavior of another operator rather than as a standalone operator. However, assuming they share the same code, I figure linking is the best way to answer the user's possible question of "What are the details on how this `where` clause is used?"

Reviewer: I do think this is a bit confusing because this is a clause, not an operator. How about something like this?

```
Each aggregate function may be optionally followed by a `where` clause, which
applies a Boolean expression that indicates, for each input value,
whether to deliver it to that aggregate. (`where` clauses are analogous
to the [`where` operator](where.md).)
```

applies a Boolean expression that indicates, for each input value,
whether to deliver it to that aggregate function.

The output field names for each aggregate and each key are optional.  If omitted,
a field name is inferred from each right-hand side, e.g, the output field for the [`count`](../aggregates/count.md)

> aggregate function is simply `count`.

Author: I chose to reference `count` since it's one more opportunity to avoid any new user confusion about `summarize` vs. `sum`.

### Examples

Average the input sequence:
```mdtest-command

> echo '1 2 3 4' | zq -z 'summarize avg(this)' -

Author: Once again, I chose a different aggregate function so we'd not be _leading_ with `summarize sum` and perhaps confuse new users.

```
=>
```mdtest-output
{avg:2.5}
```

> Sum the input sequence, leaving out the `summarize` keyword:

Author: I figured that since we used the explicit `summarize` in one example and call attention here to intentionally leaving it out, now we're free to leave it out of all the other examples.
